pkg/clusters/addons/kong/addon.go

```diff
@@ -134,46 +134,46 @@ func (a *Addon) Namespace() string {
+	urlStr, err := a.urlForService(ctx, cluster, types.NamespacedName{Namespace: a.namespace, Name: DefaultProxyServiceName}, DefaultProxyHTTPPort)
 	if err != nil {
 		return nil, err
 	}
+	return url.Parse(urlStr)
+}
 
-	if !ready {
-		return nil, fmt.Errorf("the addon is not ready on cluster %s: non-empty unresolved objects list: %+v", cluster.Name(), waitForObjects)
+// ProxyHTTPSURL provides a routable *url.URL for accessing the Kong proxy.
+func (a *Addon) ProxyHTTPSURL(ctx context.Context, cluster clusters.Cluster) (*url.URL, error) {
+	urlStr, err := a.urlForService(ctx, cluster, types.NamespacedName{Namespace: a.namespace, Name: DefaultProxyServiceName}, DefaultProxyTLSServicePort)
+	if err != nil {
+		return nil, err
 	}
-
-	return urlForService(ctx, cluster, types.NamespacedName{Namespace: a.namespace, Name: DefaultProxyServiceName}, DefaultProxyHTTPPort)
+	return url.Parse(urlStr)
 }
 
 // ProxyAdminURL provides a routable *url.URL for accessing the Kong Admin API.
 func (a *Addon) ProxyAdminURL(ctx context.Context, cluster clusters.Cluster) (*url.URL, error) {
-	waitForObjects, ready, err := a.Ready(ctx, cluster)
+	urlStr, err := a.urlForService(ctx, cluster, types.NamespacedName{Namespace: a.namespace, Name: DefaultAdminServiceName}, DefaultAdminServicePort)
 	if err != nil {
 		return nil, err
 	}
-
-	if !ready {
-		return nil, fmt.Errorf("the addon is not ready on cluster %s, see: %+v", cluster.Name(), waitForObjects)
-	}
-
-	return urlForService(ctx, cluster, types.NamespacedName{Namespace: a.namespace, Name: DefaultAdminServiceName}, DefaultAdminServicePort)
+	return url.Parse(urlStr)
 }
 
 // ProxyUDPURL provides a routable *url.URL for accessing the default UDP service for the Kong Proxy.
-func (a *Addon) ProxyUDPURL(ctx context.Context, cluster clusters.Cluster) (*url.URL, error) {
-	waitForObjects, ready, err := a.Ready(ctx, cluster)
-	if err != nil {
-		return nil, err
-	}
+func (a *Addon) ProxyUDPURL(ctx context.Context, cluster clusters.Cluster) (string, error) {
+	return a.urlForService(ctx, cluster, types.NamespacedName{Namespace: a.namespace, Name: DefaultUDPServiceName}, DefaultUDPServicePort)
+}
 
-	if !ready {
-		return nil, fmt.Errorf("the addon is not ready on cluster %s, see: %+v", cluster.Name(), waitForObjects)
-	}
+// ProxyTCPURL provides a routable *url.URL for accessing the default TCP service for the Kong Proxy.
+func (a *Addon) ProxyTCPURL(ctx context.Context, cluster clusters.Cluster) (string, error) {
+	// TCP port is exposed on the same service as the HTTP port.
+	return a.urlForService(ctx, cluster, types.NamespacedName{Namespace: a.namespace, Name: DefaultProxyServiceName}, DefaultTCPServicePort)
+}
 
-	return urlForService(ctx, cluster, types.NamespacedName{Namespace: a.namespace, Name: DefaultUDPServiceName}, DefaultUDPServicePort)
+func (a *Addon) ProxyTLSURL(ctx context.Context, cluster clusters.Cluster) (string, error) {
```

Comment:
Can we add a godoc for this func as well to stay consistent with other functions in here?